backend/company-agent/src/main/java/org/hyperledger/oa/controller/AdminController.java

```diff
@@ -130,4 +144,60 @@
         return HttpResponse.ok(config);
     }
 
+    /**
+     * Trigger the backend to write configured endpoints to the ledger. TAA digest
+     * has to be passed to explicitly confirm TTA acceptance of the user for this
+     * ledger interaction.
+     * 
+     * @param tAADigest TAA digest retrieved from "/taa/get"
+     * @return {@link HttpResponse}
+     */
+    @Post("/taa/writeEndpoints")
+    public HttpResponse<SchemaAPI> writeEndpoints(@Body String tAADigest) {
+        if (endpointService.isPresent()) {
+            try {
+                Optional<TAAInfo> taa = ac.ledgerTaa();
+                if (!taa.isEmpty()) {
+                    // do TAA acceptance even if not required or if already done
+                    // (assuming the user accepted the TAA for this single ledger interaction or
+                    // this session)
+                    TAARecord taaRecord = taa.get().getTaaRecord();
+                    if (tAADigest.equals(taaRecord.getDigest())) {
+                        ac.ledgerTaaAccept(TAAAccept.builder()
+                                // TODO use suitable mechanism, e.g. "session type".For the time being we just
+                                // use a random one.
+                                .mechanism(taa.get().getAmlRecord().getAml().keySet().iterator().next())
```

Comment:
Are you sure that you will never get any Null values on this path?

Reply:
hmm , good question. maybe I better explicitly check for null